### ISSUE TITLE
WiC broken in demo.conf

### ISSUE THREAD

**Issue by sleepinyourhat** [nyu-mll/jiant#760]
WiC works in superglue-bow.conf, but gives strange errors in demo.conf. It should either run in this setup, or else give a helpful error that tells you what needs to be set:

```
python main.py --config config/demo.conf --overrides "pretrain_tasks = wic"
```

```
06/26 02:48:09 PM: Beginning training with stopping criteria based on metric: wic_accuracy
06/26 02:48:09 PM: Fatal error in main():
Traceback (most recent call last):
  File "main.py", line 590, in <module>
    main(sys.argv[1:])
  File "main.py", line 462, in main
    phase="pretrain",
  File "/Users/srbowman/jiant/src/trainer.py", line 594, in train
    output_dict = self._forward(batch, task=task)
  File "/Users/srbowman/jiant/src/trainer.py", line 1035, in _forward
    model_out = self._model.forward(task, tensor_batch)
  File "/Users/srbowman/jiant/src/models.py", line 808, in forward
    out = self._pair_sentence_forward(batch, task, predict)
  File "/Users/srbowman/jiant/src/models.py", line 936, in _pair_sentence_forward
    logits = classifier(sent1, sent2, mask1, mask2, [batch["idx1"]], [batch["idx2"]])
  File "/Users/srbowman/miniconda3/envs/jiant/lib/python3.6/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/Users/srbowman/jiant/src/modules/simple_modules.py", line 167, in forward
    s2_ctx_emb = s2.gather(dim=1, index=idx)
RuntimeError: Invalid index in gather at ../aten/src/TH/generic/THTensorEvenMoreMath.cpp:456
Traceback (most recent call last):
  File "main.py", line 601, in <module>
    raise e  # re-raise exception, in case debugger is attached.
  File "main.py", line 590, in <module>
    main(sys.argv[1:])
  File "main.py", line 462, in main
    phase="pretrain",
  File "/Users/srbowman/jiant/src/trainer.py", line 594, in train
    output_dict = self._forward(batch, task=task)
  File "/Users/srbowman/jiant/src/trainer.py", line 1035, in _forward
    model_out = self._model.forward(task, tensor_batch)
  File "/Users/srbowman/jiant/src/models.py", line 808, in forward
    out = self._pair_sentence_forward(batch, task, predict)
  File "/Users/srbowman/jiant/src/models.py", line 936, in _pair_sentence_forward
    logits = classifier(sent1, sent2, mask1, mask2, [batch["idx1"]], [batch["idx2"]])
  File "/Users/srbowman/miniconda3/envs/jiant/lib/python3.6/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/Users/srbowman/jiant/src/modules/simple_modules.py", line 167, in forward
    s2_ctx_emb = s2.gather(dim=1, index=idx)
RuntimeError: Invalid index in gather at ../aten/src/TH/generic/THTensorEvenMoreMath.cpp:456
```

**Comment by W4ngatang** [comment]
@sleepinyourhat can you checkout `wic-v2` and re-run? I'm not seeing this now.
FWIW I ran `source user_config.sh && python -m ipdb main.py --config config/demo.conf --overrides "pretrain_tasks = wic, d_hid = 128"`

**Comment by sleepinyourhat** [comment]
```python main.py --config config/demo.conf --overrides "pretrain_tasks = wic, d_hid = 128"```

In an old experiment dir: 
```
06/26 06:15:02 PM: Beginning training with stopping criteria based on metric: wic_accuracy
06/26 06:15:03 PM: Fatal error in main():
Traceback (most recent call last):
  File "main.py", line 590, in <module>
    main(sys.argv[1:])
  File "main.py", line 462, in main
    phase="pretrain",
  File "/Users/srbowman/jiant-public/src/trainer.py", line 594, in train
    output_dict = self._forward(batch, task=task)
  File "/Users/srbowman/jiant-public/src/trainer.py", line 1035, in _forward
    model_out = self._model.forward(task, tensor_batch)
  File "/Users/srbowman/jiant-public/src/models.py", line 808, in forward
    out = self._pair_sentence_forward(batch, task, predict)
  File "/Users/srbowman/jiant-public/src/models.py", line 936, in _pair_sentence_forward
    logits = classifier(sent1, sent2, mask1, mask2, [batch["idx1"]], [batch["idx2"]])
  File "/Users/srbowman/miniconda3/envs/jiant/lib/python3.6/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/Users/srbowman/jiant-public/src/modules/simple_modules.py", line 167, in forward
    s2_ctx_emb = s2.gather(dim=1, index=idx)
RuntimeError: Invalid index in gather at ../aten/src/TH/generic/THTensorEvenMoreMath.cpp:456
Traceback (most recent call last):
  File "main.py", line 601, in <module>
    raise e  # re-raise exception, in case debugger is attached.
  File "main.py", line 590, in <module>
    main(sys.argv[1:])
  File "main.py", line 462, in main
    phase="pretrain",
  File "/Users/srbowman/jiant-public/src/trainer.py", line 594, in train
    output_dict = self._forward(batch, task=task)
  File "/Users/srbowman/jiant-public/src/trainer.py", line 1035, in _forward
    model_out = self._model.forward(task, tensor_batch)
  File "/Users/srbowman/jiant-public/src/models.py", line 808, in forward
    out = self._pair_sentence_forward(batch, task, predict)
  File "/Users/srbowman/jiant-public/src/models.py", line 936, in _pair_sentence_forward
    logits = classifier(sent1, sent2, mask1, mask2, [batch["idx1"]], [batch["idx2"]])
  File "/Users/srbowman/miniconda3/envs/jiant/lib/python3.6/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/Users/srbowman/jiant-public/src/modules/simple_modules.py", line 167, in forward
    s2_ctx_emb = s2.gather(dim=1, index=idx)
RuntimeError: Invalid index in gather at ../aten/src/TH/generic/THTensorEvenMoreMath.cpp:456
```

In a fresh dir:
```
06/26 06:15:37 PM: 	Creating task wic from scratch.
06/26 06:15:37 PM: Fatal error in main():
Traceback (most recent call last):
  File "main.py", line 590, in <module>
    main(sys.argv[1:])
  File "main.py", line 422, in main
    pretrain_tasks, target_tasks, vocab, word_embs = build_tasks(args)
  File "/Users/srbowman/jiant-public/src/preprocess.py", line 282, in build_tasks
    tasks, pretrain_task_names, target_task_names = get_tasks(args)
  File "/Users/srbowman/jiant-public/src/preprocess.py", line 466, in get_tasks
    task = _get_task(name, args, data_path=data_path, scratch_path=scratch_path)
  File "/Users/srbowman/jiant-public/src/preprocess.py", line 423, in _get_task
    task.load_data()
  File "/Users/srbowman/jiant-public/src/tasks/tasks.py", line 2300, in load_data
    self.train_data_text = _load_split(os.path.join(self.path, "train.jsonl"))
  File "/Users/srbowman/jiant-public/src/tasks/tasks.py", line 2287, in _load_split
    sent1, sent1[row["start1"] : row["end1"]]
KeyError: 'start1'
Traceback (most recent call last):
  File "main.py", line 601, in <module>
    raise e  # re-raise exception, in case debugger is attached.
  File "main.py", line 590, in <module>
    main(sys.argv[1:])
  File "main.py", line 422, in main
    pretrain_tasks, target_tasks, vocab, word_embs = build_tasks(args)
  File "/Users/srbowman/jiant-public/src/preprocess.py", line 282, in build_tasks
    tasks, pretrain_task_names, target_task_names = get_tasks(args)
  File "/Users/srbowman/jiant-public/src/preprocess.py", line 466, in get_tasks
    task = _get_task(name, args, data_path=data_path, scratch_path=scratch_path)
  File "/Users/srbowman/jiant-public/src/preprocess.py", line 423, in _get_task
    task.load_data()
  File "/Users/srbowman/jiant-public/src/tasks/tasks.py", line 2300, in load_data
    self.train_data_text = _load_split(os.path.join(self.path, "train.jsonl"))
  File "/Users/srbowman/jiant-public/src/tasks/tasks.py", line 2287, in _load_split
    sent1, sent1[row["start1"] : row["end1"]]
KeyError: 'start1'
```

**Comment by W4ngatang** [comment]
Oh! I forgot to give you the updated data.

See Slack.

**Comment by sleepinyourhat** [comment]
Trying now. In any case, look for somewhere to add an assert that'll prevent people from using new code on old data.

**Comment by sleepinyourhat** [comment]
Confirmed: #764 fixes this when used with the new data file.

**Comment by sleepinyourhat** [comment]
@W4ngatang - Mind adding the assert? I'm not sure what the issue was, so it'll take some fiddling to do it myself.

**Comment by W4ngatang** [comment]
Should be addressed by #764

**Comment by W4ngatang** [comment]
Resolved with #764
